### ISSUE TITLE
after-startup-command seems to ignore any configuration upon startup

### ISSUE THREAD

**Issue by ninicdennis** [nikitabobko/AeroSpace#556]
Hello! I am trying to configure aerospace to start both sketchybar and borders with my configurations. The docs specify to do the following:

```
after-startup-command = [
    # JankyBorders has a built-in detection of already running process,
    # so it won't be run twice on AeroSpace restart
    'exec-and-forget borders active_color=0xffe1e3e4 inactive_color=0xff494d64 width=5.0'
]
```
```
after-startup-command = ['exec-and-forget sketchybar']

# Notify Sketchybar about workspace change
exec-on-workspace-change = ['/bin/bash', '-c',
    'sketchybar --trigger aerospace_workspace_change FOCUSED_WORKSPACE=$AEROSPACE_FOCUSED_WORKSPACE'
]
```
I have done both of the above and neither will run unless I provide the absolute path like so:

```
after-startup-command = [
  'exec-and-forget /usr/local/bin/sketchybar'
  'exec-and-forget /usr/local/bin/borders'
]
```

I have a config for each of these which should work, but it seems to lose any context of the configuration when ran. I can run these both individually as services, and they work fine. But Aerospace seems to lose that context. 

## Checklist

- [x] I've searched for duplicates. My search queries were (please don't forget to search in closed issues too):
- after-startup-command
- enviornment not loading
- startup not working
- [x] I've read the guide https://nikitabobko.github.io/AeroSpace/guide
- [x] I've searched in the documentation (https://nikitabobko.github.io/AeroSpace/guide & https://nikitabobko.github.io/AeroSpace/commands). My search queries were:
- after-startup-command
- whatever the env part is in the guide

```
<!-- Replace with `aerospace --version` output -->
```

**Comment by nikitabobko** [comment]
Do you run intel mac or apple silicon mac? How did you install sketchybar and jankyborders? (is it brew, nix, macports, or smth else)

Your workaround is:

```toml
[exec.env-vars]
PATH = '/usr/local/bin/:${PATH}'
```

**Comment by ninicdennis** [comment]
@nikitabobko It is an intel mac! Both sketchybar and jankyborders was done with brew! Ill adjust the env vars later today :+1:

**Comment by MarlieChiller** [comment]
👋 Experiencing similar. I have installed sketchy and janky via nix. Running either from the cli works but when running from aerospace's`after-startup-command` seems to have no effect, even when the `PATH` is set as mentioned above. 🤔

**Comment by ellie-feng** [comment]
i actually have the opposite issue. i followed the instructions on the 'goodness' page to integrate sketchybar spaces with aerospace, but i'm not going to use sketchybar anymore so deleted what I added. restarted aerospace and now the bar shows everytime even though i've deleted everything from my toml file and uninstalled sketchybar.
![Uploading Screenshot 2024-10-07 at 18.40.17.png…]()

**Comment by junereycasuga** [comment]
i am having the same issue. I have also installed sketchybar and jankyborders using nix. there were no problems before i moved to use nix

**Comment by junereycasuga** [comment]
Update:

Updating the startup command like the following works for me
```
after-startup-command = [
  'exec-and-forget /run/current-system/sw/bin/sketchybar',
  'exec-and-forget /run/current-system/sw/bin/borders active_color=0xffe1e3e4 inactive_color=0xff494d64 width=6.0 blacklist="iPhone Mirroring"'
]
```

**Comment by MarlieChiller** [comment]
That doesnt seem to work for me @junereycasuga. I seem to have a different bin path for borders. If I specify that path in my `after-startup-command` I still have no joy.
```
> which borders                                                                                                                                                          
/Users/{USERNAME}/.nix-profile/bin/borders

**Comment by Mattchine** [comment]
I faced a similar problem, but I think I found a workaround.
In my case, the workspace doesn’t automatically show in the bar on startup. However, after manually running `sketchybar --reload`, it seems to work fine.

So I added the flag to the startup command, which solved the problem.
```bash
after-startup-command = [
	'exec-and-forget sketchybar --reload', # add reload flag
	'exec-and-forget borders active_color=0xffd79921 inactive_color=0xff282828 width=5.0'
]
```

**Comment by MarlieChiller** [comment]
Update: the latest version seems to have fixed this for me. For reference my `after-startup-command` is:

```
  'exec-and-forget /Users/{USERNAME}/.nix-profile/bin/borders active_color=0xffe1e3e4 inactive_color=0xff494d64 width=5.0'
]

**Comment by gnohj** [comment]
The latest update partially fixed the issue for me (was on `0.14.2`). The `after-startup-command` still doesn't fully work for Sketchy – it doesn't render all the menu items right away, but it does work for the borders. It’s only after I trigger an on-focus change that everything shows up. I’m okay with this though—at least I don’t have to run Sketchy directly in the terminal anymore.

```
aerospace CLI client version: 0.15.2-Beta b6cf82771f245ab7349a93baf8709e171537ff58
AeroSpace.app server version: 0.15.2-Beta b6cf82771f245ab7349a93baf8709e171537ff58

after-startup-command = [
  'exec-and-forget sketchybar --reload',
  'exec-and-forget /opt/homebrew/opt/borders/bin/borders active_color=0xCFEBBF83 inactive_color=0x33434c5e width=4.0'
]

on-focus-changed = [
  'move-mouse window-lazy-center',
  'exec-and-forget /bin/bash -c /opt/homebrew/bin/sketchybar --trigger front_app_switched',
  'exec-and-forget sketchybar --trigger update_windows'
]
```

**Comment by zuqini** [comment]
Also encountered this issue on my Intel Mac.

Running `aerospace list-exec-env-vars` shows that the default PATH is defined as:
```
PATH=/opt/homebrew/bin:/opt/homebrew/sbin:/usr/bin:/bin:/usr/sbin:/sbin
```

This is only an issue for Intel Macs because [brew installs to /opt/homebrew/bin for Apple Silicon Macs but /usr/local/bin for Intel Macs](https://docs.brew.sh/Installation#1), and in this case, sketchybar is installed to `/usr/local/bin/` which is not part of the `PATH` by default.

```
$ which sketchybar
/usr/local/bin/sketchybar
```

The workaround mentioned [here](https://github.com/nikitabobko/AeroSpace/issues/556#issuecomment-2392419701) fixes the `PATH` on the Intel Mac, but setting it removes the `/opt/homebrew` paths, which unfortunately then breaks the `PATH` on Apple Silicon.
```
PATH=/usr/local/bin/:/usr/bin:/bin:/usr/sbin:/sbin
```

**Comment by icecreammachine** [comment]
> Do you run intel mac or apple silicon mac? How did you install sketchybar and jankyborders? (is it brew, nix, macports, or smth else)
> 
> Your workaround is:
> 
> ```toml
> [exec.env-vars]
> PATH = '/usr/local/bin/:${PATH}'
> ```

When I add this to the config file I get some error messages, do I need to change something else? I am using intel Mac.

```
exec.env-vars.accordion-padding: Expected type is 'string'. But actual type is 'integer'

exec.env-vars.automatically-unhide-macos-hidden-apps: Expected type is 'string'. But actual type is 'bool'

exec.env-vars.enable-normalization-flatten-containers: Expected type is 'string'. But actual type is 'bool'

exec.env-vars.enable-normalization-opposite-orientation-for-nested-containers: Expected type is 'string'. But actual type is 'bool'

exec.env-vars.on-focused-monitor-changed: Expected type is 'string'. But actual type is 'array'

exec.env-vars.start-at-login: Expected type is 'string'. But actual type is 'bool'
```

**Comment by hajiboy95** [comment]
> Update:
> 
> Updating the startup command like the following works for me
> 
> ```
> after-startup-command = [
>   'exec-and-forget /run/current-system/sw/bin/sketchybar',
>   'exec-and-forget /run/current-system/sw/bin/borders active_color=0xffe1e3e4 inactive_color=0xff494d64 width=6.0 blacklist="iPhone Mirroring"'
> ]
> ```
> 
> although one issue i found with this is that my sketchybar doesn't seem to be working well. when executing sketchybar independently, it seems like it properly loads my config just fine. but if sketchybar was started by aerospace, i only get an empty bar (even if i specify which config sketchybar needs to use)

got exactly the same problem just doesn't seem to be working at all. for me sketchybar just doesn't appear if i try this approach. already tried all the solutions mentioned here
